agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -4795,6 +4797,13 @@ bool slave_thread::channel_selection_get_channel_preference(ieee1905_1::CmduMess
                     return false;
                 }
 
+                // Check if channel is valid for operating class
+                if (op_class_chan_set.find(*channel) == op_class_chan_set.end()) {
+                    LOG(ERROR) << "Channel " << +*channel << " invalid for operating class "
+                               << +operating_class;
+                    return false;
+                }
+
                 ss << int(*channel);
```

Comment:
[I guess (line 10)](https://cppinsights.io/s/2fe9eb81) the same trick will work here too, but need to verify:
```suggestion
                ss << +(*channel);
```

Reply:
Yes it would. However, with #1392 that wouldn't even be needed.

Regardless, this is code that is not changed by this PR so I'm not touching it :-)